### PR TITLE
Add Sparse MaxPool3D

### PR DESCRIPTION
**Pull request by zkh2016**
### PR types

New features
### PR changes

APIs
### Describe

1. 新增`paddle.sparse.MaxPool3D`
2. 新增`paddle.sparse.functional.max_pool3d`
3. sparse的max_pool3d和dense的定义相同，只是输入是SparseCooTensor
4. 当前只支持SparseCooTensor，layout=NDHWC
5. 修复to_dense得到错误layout的bug

英文文档：
image
image

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_sparse_pool_op.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+from __future__ import print_function
+import unittest
+import numpy as np
+import paddle
+import paddle.fluid.core as core
+from paddle import _C_ops
+from paddle.fluid.framework import _test_eager_guard
+
+
+class TestMaxPool3DFunc(unittest.TestCase):
```
Comment (xingfeng01) [inline python/paddle/fluid/tests/unittests/test_sparse_pool_op.py]: 后续增加一些测试

**Review by xingfeng01 — DISMISSED** [review]

File: python/paddle/sparse/functional/pool.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+def max_pool3d(x,
+               kernel_size,
+               stride=None,
+               padding=0,
+               ceil_mode=False,
+               data_format="NDHWC",
+               name=None):
+    """
+    This API implements sparse max pooling 3d operation.
```
Comment (TCChenlong) [inline python/paddle/sparse/functional/pool.py]: ~This API~

File: python/paddle/sparse/functional/pool.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    See more details in :ref:`api_sparse_pooling_MaxPool3d` .
+    Args:
```
Comment (TCChenlong) [inline python/paddle/sparse/functional/pool.py]: Please add a blank line before Args

File: python/paddle/sparse/functional/pool.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                        The default is `"NCDHW"`. When it is `"NCDHW"`, the data is stored in the order of:
+                        `[batch_size, input_channels, input_depth, input_height, input_width]`. Currently only support `"NDHWC"` .
+        name(str, optional): For detailed information, please refer
+                             to :ref:`api_guide_Name`. Usually name is no need to set and
+                             None by default.
+    
+    Returns:
+        Tensor: The output tensor of pooling result. The data type is same as input tensor.
+    
+    Raises:
```
Comment (TCChenlong) [inline python/paddle/sparse/functional/pool.py]: Delete Raises

File: python/paddle/sparse/functional/pooling.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+def max_pool3d(x,
+               kernel_size,
+               stride=None,
+               padding=0,
+               ceil_mode=False,
+               data_format="NDHWC",
+               name=None):
+    """
+    This API implements sparse max pooling 3d operation.
```
Comment (TCChenlong) [inline python/paddle/sparse/functional/pooling.py]: As above

File: python/paddle/sparse/layer/pool.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+          The data type is same as input x.
+
+    Examples:
+        .. code-block:: python
+
+            import paddle
+
+            # max pool3d
+            input = paddle.randn((1, 3, 32, 32, 2))
+            MaxPool3D = nn.MaxPool3D(kernel_size=2,
```
Comment (TCChenlong) [inline python/paddle/sparse/layer/pool.py]: Please use `paddle.sparse.MaxPool3D`

File: python/paddle/sparse/functional/pooling.py [hunk 7]
```diff
@@ -0,0 +1,101 @@
+#   Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
```
Comment (TCChenlong) [inline python/paddle/sparse/functional/pooling.py:1]: rm this file？

File: python/paddle/sparse/layer/pooling.py [hunk 8]
```diff
@@ -0,0 +1,105 @@
+# Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
```
Comment (TCChenlong) [inline python/paddle/sparse/layer/pooling.py:1]: rm this file?

File: python/paddle/fluid/tests/unittests/test_sparse_pool_op.py (hunk 1 above)
Comment (zkh2016) [reply python/paddle/fluid/tests/unittests/test_sparse_pool_op.py]: ok

File: python/paddle/sparse/functional/pool.py (hunk 2 above)
Comment (zkh2016) [reply python/paddle/sparse/functional/pool.py]: done

File: python/paddle/sparse/functional/pool.py (hunk 3 above)
Comment (zkh2016) [reply python/paddle/sparse/functional/pool.py]: done

File: python/paddle/sparse/functional/pool.py (hunk 4 above)
Comment (zkh2016) [reply python/paddle/sparse/functional/pool.py]: done

File: python/paddle/sparse/functional/pooling.py (hunk 5 above)
Comment (zkh2016) [reply python/paddle/sparse/functional/pooling.py]: done

File: python/paddle/sparse/layer/pool.py (hunk 6 above)
Comment (zkh2016) [reply python/paddle/sparse/layer/pool.py]: 已删除pool.py

File: python/paddle/sparse/functional/pooling.py (hunk 7 above)
Comment (zkh2016) [reply python/paddle/sparse/functional/pooling.py:1]: 已留pooling.py，删pool.py

**Review by TCChenlong — APPROVED** [review]
LGTM for API docs

**Review by xingfeng01 — APPROVED** [review]

**Review by lanxianghit — APPROVED** [review]

**Pull request merged**